### ISSUE TITLE
KeyError: 'CONTENT_LENGTH' on empty request with s2repoze

### ISSUE THREAD

**Issue by adborden** [IdentityPython/pysaml2#676]
On GET /, we're seeing the application crash with a KeyError. It looks like the boolean condition is wrong.

## Code Version
v4.9.0
repoze.who 2.0

## Expected Behavior
Request should go through successfully without an identity.

## Current Behavior
```
mod_wsgi (pid=26207): Exception occurred processing WSGI script '/etc/ckan/apache.wsgi'.
Traceback (most recent call last):
  File "/usr/lib/ckan/lib/python2.7/site-packages/paste/cascade.py", line 130, in __call__
    return self.apps[-1](environ, start_response)
  File "/usr/lib/ckan-new/src/ckan/ckan/config/middleware.py", line 226, in __call__
    return self.app(environ, start_response)
  File "/usr/lib/ckan/lib/python2.7/site-packages/paste/registry.py", line 379, in __call__
    app_iter = self.application(environ, start_response)
  File "/usr/lib/ckan/lib/python2.7/site-packages/repoze/who/middleware.py", line 76, in __call__
    identity = api.authenticate()
  File "/usr/lib/ckan/lib/python2.7/site-packages/repoze/who/api.py", line 129, in authenticate
    ids = self._identify()
  File "/usr/lib/ckan/lib/python2.7/site-packages/repoze/who/api.py", line 308, in _identify
    identity = plugin.identify(self.environ)
  File "/usr/lib/ckan-new/src/pysaml2/src/saml2/s2repoze/plugins/sp.py", line 478, in identify
    has_content_length = "CONTENT_LENGTH" in environ or environ["CONTENT_LENGTH"]
KeyError: 'CONTENT_LENGTH'
```

## Possible Solution
It looks like the `or` should be flipped to an `and` to avoid the key error.

## Steps to Reproduce
1. Configure pysaml2 with repoze.who, who.ini:
```
[plugin:auth_tkt]
use = ckan.lib.auth_tkt:make_plugin
secret = ...

[plugin:saml2auth]
use = saml2.s2repoze.plugins.sp:make_plugin
saml_conf = ckanext.saml2.config_max.sp_config
remember_name = auth_tkt
sid_store = /var/tmp/ckan/outstanding
identity_cache = memcached

[plugin:saml2_challenge_decider]
use = saml2.s2repoze.plugins.challenge_decider:make_plugin
path_login = /user/login

[general]
request_classifier = repoze.who.classifiers:default_request_classifier
challenge_decider = saml2_challenge_decider

[identifiers]
plugins =
    saml2auth
    auth_tkt

[authenticators]
plugins =
    saml2auth

[challengers]
plugins =
    saml2auth
```
2. Issue `curl localhost`

**Comment by c00kiemon5ter** [comment]
Thanks for filling in this issue. To be honest, I don't think that people are actually using the `s2repoze` module. Nonetheless, this is now fixed by 0f29cc4931e322f8f5ec27c551d1ded463b6789f

**Comment by adborden** [comment]
For what it's worth, any [ckan](https://ckan.org/) instance with [SAML2 authentication](https://github.com/GSA/ckanext-saml2/network/members) is based on the s2repoze module.
